app/js/app/app.js

```diff
@@ -700,9 +700,17 @@ define([
 
                             if (ns.length > 0) {
 
-                                $rootScope.notificationDoc = ns[0];
+                                // dirty hack for student/teacher questionnaires, but notifications wont be like this for long (05/17)
+                                for (var i = 0; i < ns.length; i++) {
 
-                                $rootScope.modals.notification.show();
+                                    if ($rootScope.user.role.toLowerCase() == ns[i].tags[0]) {
```

Comment:
This can only cope with single tagged things; we couldn't make something that was tagged `teacher` and `student` this way. I think this is acceptable, since we're likely to ditch this endpoint soon anyway - but worth noting!